### PR TITLE
Adds support from reads

### PR DESCRIPTION
**Pull request by sk-sahu**
Adds two things - 
* SRA/GTEX support
* Manually fastq.gz files input

Tried to run - 
```bash
$ ./nextflow run main.nf -profile docker,test_with_sra
N E X T F L O W  ~  version 20.07.1
Launching `main.nf` [berserk_sinoussi] - revision: 5d55c6f508
executor >  local (7)
[e5/047316] process > get_accession (SRR8180445)                     [100%] 1 of 1 ✔
[86/76ef4b] process > fastqc (SRR8180445)                            [100%] 1 of 1 ✔
[aa/01a8ea] process > trimmomatic (SRR8180445)                       [100%] 1 of 1 ✔
[d7/73a0ae] process > fastqc_trimmed (SRR8180445)                    [100%] 1 of 1 ✔
[ca/18abf9] process > bwa_index (On human_g1k_v37_decoy.small.fasta) [100%] 1 of 1 ✔
[c3/da2dc1] process > bwa_mem (SRR8180445)                           [100%] 1 of 1 ✔
[bd/6dfc0e] process > polysolver (SRR8180445)                        [100%] 1 of 1, failed: 1 ✘
```
<details>
  <summary>Error - Click to expand!</summary>

```bash
Error executing process > 'polysolver (SRR8180445)'

Caused by:
  Process `polysolver (SRR8180445)` terminated with an error exit status (1)

Command executed:

  bash /home/polysolver/scripts/shell_call_hla_type       SRR8180445_aligned.bam       Unknown       1       hg19       ILMFQ       0       SRR8180445

Command exit status:
  1

Command output:
  -rw-r--r-- 1 root root 35K Oct 15 18:59 SRR8180445/temp.checkpairs
  99
  65
  65
  147
  99
  147
  83
  163
  99
  147
  temp.checkpairs # lines = 10000 SRR8180445/temp.checkpairs
  -rw-r--r-- 1 root root 2 Oct 15 18:59 SRR8180445/check.status.out.txt
  check_bam_flag_pairs status = 1
  skipping insert size distribution
  getting matching tags
  getting chr6 region
  build=hg19

Command error:
  [samopen] SAM header is present: 13 sequences.
  [Thu Oct 15 18:59:52 UTC 2020] picard.sam.SamToFastq INPUT=SRR8180445/tag.bam FASTQ=SRR8180445/tag.1.fastq SECOND_END_FASTQ=SRR8180445/tag.2.fastq VALIDATION_STRINGENCY=SILENT    OUTPUT_PER_RG=false RE_REVERSE=true INTERLEAVE=false INCLUDE_NON_PF_READS=false READ1_TRIM=0 READ2_TRIM=0 INCLUDE_NON_PRIMARY_ALIGNMENTS=false VERBOSITY=INFO QUIET=false COMPRESSION_LEVEL=5 MAX_RECORDS_IN_RAM=500000 CREATE_INDEX=false CREATE_MD5_FILE=false
  [Thu Oct 15 18:59:52 UTC 2020] Executing as root@9131c189ef7e on Linux 4.9.51-10.52.amzn1.x86_64 amd64; OpenJDK 64-Bit Server VM 1.7.0_79-b14; Picard version: 1.731(c12a4a98bc3c4f8333565557910d96b939985895_1402405046) JdkDeflater
  [Thu Oct 15 18:59:53 UTC 2020] picard.sam.SamToFastq done. Elapsed time: 0.01 minutes.
  Runtime.totalMemory()=505413632
  [bam_parse_region] fail to determine the sequence name.
  [main_samview] region "6:29909037-29913661" specifies an unknown reference name. Continue anyway.
  [bam_parse_region] fail to determine the sequence name.
  [main_samview] region "6:31321649-31324964" specifies an unknown reference name. Continue anyway.
  [bam_parse_region] fail to determine the sequence name.
  [main_samview] region "6:31236526-31239869" specifies an unknown reference name. Continue anyway.
  [samopen] SAM header is present: 13 sequences.
  [sam_read1] reference 'ID:samtools.1  PN:samtools     PP:samtools     VN:1.11 CL:samtools sort -@ 1 -o SRR8180445_aligned.bam
  180445_trimmed_R2.fastq.gz
  ' is recognized as '*'.
  [main_samview] truncated file.
```

</details>

This error seems due to lack of proper test data. But functional wise the process works fine.

I'll run again with whole genome as reference. Meanwhile if code wise review can be done!!

### CODE REVIEW

File: nextflow.config [hunk 1]
```diff
@@ -5,23 +5,47 @@ manifest {
...
+    insert_calc   = 0
 
     // pipeline inputs
-    outdir      = 'results'
-    help        = false
+    outdir        = 'results'
+    help          = false
+
+    // Max resources
+    max_memory     = 760.GB
```
Comment (cgpu) [inline nextflow.config:30]: 👀

**Review by cgpu — APPROVED** [review]
Looks good 👍

File: main.nf [hunk 2]
```diff
@@ -50,6 +102,174 @@ if (params.bams) {
...
+    file ("logs/${name}_trimmomatic.log") into ch_trimmomatic_logs
+
+    script:
+    mode = single_end ? 'SE' : 'PE'
+    out = single_end ? "${name}_trimmed.fastq.gz" : "${name}_trimmed_R1.fastq.gz ${name}_unpaired_R1.fastq.gz ${name}_trimmed_R2.fastq.gz ${name}_unpaired_R2.fastq.gz"
+    output_filename = single_end ? "${name}_trimmed.fastq.gz" : "${name}_trimmed_R{1,2}.fastq.gz"
+    keepbothreads = single_end == true ? '' : ':2:true'
+    """
+    trimmomatic \
+      $mode \
```
Comment (cgpu) [inline main.nf:183]: :confused-parrot: w8 a minute, I thought we were swapping this? either way fine by me, trimmomatic is used across the globe with steady incidents of complaints, safe to say it's a legit bioinfo tool

File: nextflow.config (hunk 1 above)
Comment (sk-sahu) [reply nextflow.config]: Modified

**Pull request merged**